tests/testthat/test-antimalarial-resistance.R

```diff
@@ -0,0 +1,161 @@
+test_that('set_antimalarial_resistance() toggles resistance on', {
+  simparams <- get_parameters()
+  simparams <- set_drugs(parameters = simparams, drugs = list(SP_AQ_params))
+  simparams <- set_clinical_treatment(parameters = simparams,
+                                      drug = 1,
+                                      timesteps = 1,
+                                      coverages = 1)
+  set_antimalarial_resistance(parameters = simparams,
+                              drug = 1,
+                              timesteps = 1,
+                              artemisinin_resistance = 0.5,
+                              partner_drug_resistance = 0,
+                              slow_parasite_clearance_prob = 0.5,
+                              early_treatment_failure_prob = 0.6,
+                              late_clinical_failure_prob = 0.2,
+                              late_parasitological_prob = 0.3,
+                              reinfection_prob = 0.4) -> simparams
+  expect_identical(object = simparams$antimalarial_resistance, expected = TRUE)
+})
+
+test_that('set_antimalarial_resistance() errors if parameter inputs of different length to timesteps', {
+  simparams <- get_parameters()
+  simparams <- set_drugs(parameters = simparams, drugs = list(SP_AQ_params))
+  simparams <- set_clinical_treatment(parameters = simparams,
+                                      drug = 1,
+                                      timesteps = 1,
+                                      coverages = 1)
+  expect_error(object =  set_antimalarial_resistance(parameters = simparams,
+                                                     drug = 1,
+                                                     timesteps = c(1, 10),
+                                                     artemisinin_resistance = 0.5,
+                                                     partner_drug_resistance = 0,
+                                                     slow_parasite_clearance_prob = 0.5,
+                                                     early_treatment_failure_prob = 0.6,
+                                                     late_clinical_failure_prob = 0.2,
+                                                     late_parasitological_prob = 0.3,
+                                                     reinfection_prob = 0.4))
+})
+
+test_that('set_antimalarial_resistance() errors if resistance proportions outside bound of 0-1', {
+  simparams <- get_parameters()
+  simparams <- set_drugs(parameters = simparams, drugs = list(SP_AQ_params))
+  simparams <- set_clinical_treatment(parameters = simparams,
+                                      drug = 1,
+                                      timesteps = 1,
+                                      coverages = 1)
+  expect_error(object =  set_antimalarial_resistance(parameters = simparams,
```

Comment:
In general, I'd explicitly test for the error message you're expecting (like [here](https://github.com/mrc-ide/malariasimulation/blob/397a7b7efe90958dd01f97110a1d16c71d041f33/tests/testthat/test-parameter_draws.R#L3))

Otherwise its easy to past the test as a result of a differnt (and potentially unexpected/wrong) error

Reply:
I've changed the test to so that the error message matches that which is being tested in the set_antimalarial_resistance("Artemisinin and partner-drug resistance proportions must fall between 0 and 1") function:

"Artemisinin and partner-drug resistance proportions must fall between 0 and 1"